napalm/iosxr/iosxr.py

```diff
@@ -106,10 +106,10 @@ def close(self):
 
     def is_alive(self):
         """Returns a flag with the state of the connection."""
-        # Simply returns the flag from Netmiko
-        return {
-            'is_alive': self.device.device.is_alive()
-        }
+        if self.device is None:
+            return {'is_alive': False}
+        # Simply returns the flag from pyIOSXR
+        return {'is_alive': self.device.is_alive()}
```

Comment:
In `__init__` method should we just set:

self.device = None

and then not create the IOSXR object until the open() call? In other words, move lines 90 - 96 to be in the open() method.

Also should the close() method be setting `self.device = None`. I looked in the NAPALM IOS driver and in `__init__` I am setting self.device = None (and don't create the Netmiko object until open() call). I am not doing the self.close though (i.e. I think we should be setting self.device = None in close).

What do you think on this?

Reply:
I don't think it shouldn't matter whether we put do the `self.device = None` or not as long as the API behaves as documented. Would be nice to be consistent amongst devices that use netmiko though.